src/components/application_manager/test/CMakeLists.txt

```diff
@@ -74,43 +83,92 @@ set(RESUMPTION_EXCLUDE
+    # Temporary exclude some resumption tests from build due to compile errors APPLINK-26346
     list(APPEND RESUMPTION_EXCLUDE resumption_data_test.cc)
   endif()
 endif()
 
 collect_sources(RESUMPTION_SOURCES "${RESUMPTION_PATHS}" "${RESUMPTION_EXCLUDE}")
 
 list(APPEND RESUMPTION_SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/mock_message_helper.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/message_helper/mock_message_helper.cc
 )
 
+#=============== STATE CONTROLLER =====================
 set(STATE_CONTROLLER_PATHS
   ${CMAKE_CURRENT_SOURCE_DIR}/state_controller
 )
 collect_sources(STATE_CONTROLLER_SOURCES "${STATE_CONTROLLER_PATHS}")
 
 list(APPEND STATE_CONTROLLER_SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/mock_message_helper.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/message_helper/mock_message_helper.cc
+)
+
+#=============== MOBILE MESSAGE =====================
+set(MOBILE_MESSAGE_PATHS
+  ${CMAKE_CURRENT_SOURCE_DIR}/mobile_message
+)
+collect_sources(MOBILE_MESSAGE_SOURCES "${MOBILE_MESSAGE_PATHS}")
+
+#=============== APPLICATION =====================
+set(APPLICATION_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/application_impl_test.cc
+)
+
+list(APPEND APPLICATION_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/message_helper/mock_message_helper.cc
 )
 
-set(EXCLUDE_PATHS
-  ${COMMANDS_PATHS}
-  ${MESSAGE_HELPER_PATHS}
-  ${REQUEST_CONTROLLER_PATHS}
-  ${RESUMPTION_PATHS}
-  ${STATE_CONTROLLER_PATHS}
+#=============== APPLICATION_MANAGER =====================
+set(APPLICATION_MANAGER_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/application_manager_impl_test.cc
 )
 
-set(PATHS
-  ${CMAKE_CURRENT_SOURCE_DIR}/include
-  ${CMAKE_CURRENT_SOURCE_DIR}
+#=============== APPLICATION STATE =====================
+set(APPLICATION_STATE_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/application_state_test.cc
 )
-# Temporary exclude some resumption tests from build due to compile errors
-list(APPEND EXCLUDE_PATHS
-  zero_request_amount_test.cc
+
+#=============== EVENT ENGINE =====================
+set(AM_EVENT_ENGINE_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/event_engine_test.cc
+)
+
+#=============== HMI CAPABILITIES =====================
+set(AM_HMI_CAPABILITIES_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/hmi_capabilities_test.cc
 )
 
-collect_sources(SOURCES "${PATHS}" "${EXCLUDE_PATHS}")
+list(APPEND AM_HMI_CAPABILITIES_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/message_helper/mock_message_helper.cc
+)
+
+#=============== POLICY EVENT OBSERVER =====================
+set(AM_POLICY_EVENT_OBSERVER_SOURCES
+# Temporary exclude some policy event tests from build due to compile errors APPLINK-26346
+#  ${CMAKE_CURRENT_SOURCE_DIR}/policy_event_observer_test.cc
+)
+
+##=============== POLICY HANDLER =====================
+set(AM_POLICY_HANDLER_SOURCES
+# Temporary exclude some policy handler tests from build due to compile errors APPLINK-26346
+#  ${CMAKE_CURRENT_SOURCE_DIR}/policy_handler_test.cc
+)
+
+list(APPEND AM_POLICY_HANDLER_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/message_helper/mock_message_helper.cc
```

Comment:
@AGritsevich I propose you to create static library from `mock_message_helper.cc` instead of adding this source to each set